android/app/src/main/java/app/organicmaps/search/SearchResult.java

```diff
@@ -141,4 +142,37 @@ public Spannable getFormattedTitle(@NonNull Context context)
 
     return builder;
   }
+
+  public String getAddress(@NonNull Context context)
+  {
+    return description.region;
+  }
+
+  public Spannable getFormattedAddress(@NonNull Context context)
+  {
+      final String address = getAddress(context);
+      final String addr = address.toLowerCase();
+      String title = getTitle(context);
+      final SpannableStringBuilder builder = new SpannableStringBuilder(address);
+
+      if (highlightRanges != null)
```

Comment:
What is encoded/stored in highlightRanges?

Reply:
Actually highlighted ranges is coming from cpp files which have two consecutive pairs of integers first represents start index and second represents length of the substring to highlight in title string. 

Please see the function getFormattedTitle which is used to highlight the title.

I basically took those substrings from titles which are being highlighted and searched in the address and if it finds any match it highlights them.

Reply:
Shouldn't you search/match the tokens from a user's search query instead?

Reply:
No, this change only highlights the address if there is a highlight already existing in title otherwise it do not highlights anything even if text is present in query.

I am working on matching through query but unable to find the exact implementation. Should i close this pr and raise another one after dealing with query? Or do changes in this pr after some times?

Reply:
The same feature/issue should be worked in the same PR. You can apply your changes on top to be able to compare them with an older implementation.

When the PR is approved, either you can rebase/squash commits (usually if there is more than one final logical commit), or we can squash everything into one commit during the merge.

Reply:
Sure